src/UserService.Validation/User/CreateUserRequestValidator.cs

```diff
@@ -32,17 +47,28 @@ public CreateUserRequestValidator()
                 user => !string.IsNullOrEmpty(user.MiddleName),
                 () =>
                     RuleFor(user => user.MiddleName)
-                        .Must(x => !NameRegex.IsMatch(x))
-                        .WithMessage("Middle name must not contain numbers")
+                        .Must(x => !NumberRegex.IsMatch(x))
+                        .WithMessage("Middle name must not contain numbers.")
+                        .Must(x => !SpecialCharactersRegex.IsMatch(x))
+                        .WithMessage("Middle name must not contain special characters.")
                         .MaximumLength(32)
-                        .WithMessage("Middle name is too long."));
+                        .WithMessage("Middle name is too long.")
+                        .Must(x => NameRegex.IsMatch(x.Trim()))
+                        .WithMessage("Middle name contains invalid characters."));
 
             When(
                 user => !string.IsNullOrEmpty(user.City),
                 () =>
                     RuleFor(user => user.City)
                         .MaximumLength(32)
```

Comment:
why 32? use constant

Reply:
We usually write value, without variables